croissant/do.lua

```diff
@@ -263,10 +263,25 @@ local function runFile(script, arguments)
     end
 end
 
+local function getHistoryFile(debugger, mode)
```

Comment:
You cache the response so we don't stat two files each time we add something to the history

Reply:
I don't quite understand what you mean by that. Could you elaborate?

Reply:
You only need the information about which file is the right one once. So keep that information somewhere (cache it), and return that instead of checking files each time `getHistoryFile` is called.

Reply:
Where would you implement the cache? Function wise? Session wise on boot?
Implementing a cache is a delicate task that should be handled with care, with some edge cases.
I don't think checking for a file existence is a task worth caching, honestly.

In addition, is this duality between xdg files and home files a feature that you want to keep in croissant? I mean, is there a reason to keep it long term? I personally would migrate towards xdg completely, and not check for files in $HOME.
Maybe if you want to keep it customizable, a envvar could be read. I personally never use those per app config path envvars, I have $XDG_{FOLDER}_HOME if I ever want to move all my configurations